zebra-chain/src/primitives/zcash_history.rs

```diff
@@ -261,21 +263,74 @@ fn block_to_history_node(
     // There is no direct `std::primitive::u128` to `bigint::U256` conversion
     let work = bigint::U256::from_big_endian(&work.as_u128().to_be_bytes());
 
-    let sapling_tx_count = count_sapling_transactions(block);
+    let sapling_tx_count = count_sapling_transactions(block.clone());
+    let orchard_tx_count = count_orchard_transactions(block);
+
+    match network_upgrade {
+        NetworkUpgrade::Genesis
+        | NetworkUpgrade::BeforeOverwinter
+        | NetworkUpgrade::Overwinter
+        | NetworkUpgrade::Sapling
+        | NetworkUpgrade::Blossom => {
+            panic!("HistoryTree does not exist for pre-Heartwood upgrades")
+        }
```

Comment:
panic, or error?

Reply:
Can remote nodes trigger this error?
Can Zebra recover from this error by rejecting a block, or is it a programmer bug?